packages/web3-core/src/web3_request_manager.ts

```diff
@@ -274,6 +274,16 @@ export class Web3RequestManager<
 		response: JsonRpcResponse<ResultType, ErrorType>,
 		{ legacy, error }: { legacy: boolean; error: boolean },
 	): JsonRpcResponse<ResultType> | never {
+		if (isNullish(response)) {
```

Comment:
is any provider returning null for valid requests (for which some valid result is expected)?